src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -910,3 +936,91 @@ def firewall_rule_update(
         resource_group_name=resource_group_name,
         start_ip_address=start_ip_address or instance.start_ip_address,
         end_ip_address=end_ip_address or instance.end_ip_address)
+
+#####
+#           sql server key
+#####
+
+def server_key_create(
+        client,
+        resource_group_name,
+        server_name,
+        key_id=None):
+    
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.create_or_update(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name,
+        parameters=ServerKey(
+            server_key_type=ServerKeyType.azure_key_vault.value,
+            uri=key_id
+        )
+    )
+
+def server_key_get(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.get(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def server_key_delete(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.delete(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def get_server_key_name_from_uri(uri):
```

Comment:
`_get_server_key_name_from_uri` (leading underscore) to signify that it's a private helper function

Reply:
Would be nice if there was a quick comment to explain the purpose of this function e.g. "The SQL server key API requires that the server key has a specific name based on the vault, key name, and key version".